client/checkout/woopay/direct-checkout/woopay-direct-checkout.js

```diff
@@ -165,11 +165,17 @@ class WoopayDirectCheckout {
 	static redirectToWooPay( elements, useCheckoutRedirect ) {
 		elements.forEach( ( element ) => {
 			element.addEventListener( 'click', async ( event ) => {
-				event.preventDefault();
-
 				// Store href before the async call to not lose the reference.
 				const currTargetHref = event.currentTarget.href;
 
+				// If there's no link where to redirect the user, do not break the expected behavior.
+				if ( ! currTargetHref ) {
+					this.teardown();
```

Comment:
@lovo-h I've added one more commit https://github.com/Automattic/woocommerce-payments/pull/8214/commits/922425b5d4cb5ff995385ca84271d5266efb372f let me know if it makes sense to you.

When @cesarcosta99 found the issue ( https://github.com/Automattic/woopay/pull/2513#pullrequestreview-1889442228 ) I was thinking the issue could happen for various reasons. e.g. The merchant is using a custom a template that uses a different selector. In that case we shouldn't break the normal checkout flow.

Also, I'm putting the `this.teardown();` here not sure if it's OK to do that in case we end the event handler.

Reply:
@asumaran, that's a good call, and it makes sense 👍

> Also, I'm putting the `this.teardown();` here not sure if it's OK to do that in case we end the event handler.

Since we will be redirected to the Merchant checkout, the class will be automatically torn down. But in this case, it's a good practice to manually remove the listeners by calling the `teardown()` function.